batcher/aligned-sdk/src/communication/batch.rs

```diff
@@ -0,0 +1,85 @@
+use log::debug;
+
+use crate::{
+    core::{
+        errors,
+        types::{
+            AlignedVerificationData, BatchInclusionData, Chain, VerificationCommitmentBatch,
+            VerificationDataCommitment,
+        },
+    },
+    sdk::verify_proof_onchain,
+};
+
+const AWAIT_BATCH_VERIFICATION_RETRIES: u64 = 10;
+const AWAIT_BATCH_VERIFICATION_TIMEOUT: u64 = 10;
+
+pub fn handle_batch_inclusion_data(
+    batch_inclusion_data: BatchInclusionData,
+    aligned_verification_data: &mut Vec<AlignedVerificationData>,
+    verification_data_commitments_rev: &mut Vec<VerificationDataCommitment>,
+) -> Result<(), errors::SubmitError> {
+    debug!("Received response from batcher");
+    debug!(
+        "Batch merkle root: {}",
+        hex::encode(batch_inclusion_data.batch_merkle_root)
+    );
+    debug!("Index in batch: {}", batch_inclusion_data.index_in_batch);
+
+    let verification_data_commitment = verification_data_commitments_rev
+        .pop()
+        .ok_or_else(|| errors::SubmitError::EmptyVerificationDataCommitments)?;
+
+    if verify_response(&verification_data_commitment, &batch_inclusion_data) {
+        aligned_verification_data.push(AlignedVerificationData::new(
+            &verification_data_commitment,
+            &batch_inclusion_data,
+        ));
+    }
+
+    Ok(())
+}
+
+pub async fn await_batch_verification(
+    aligned_verification_data: &AlignedVerificationData,
+    rpc_url: &str,
+    chain: Chain,
+) -> Result<(), errors::SubmitError> {
+    for _ in 0..AWAIT_BATCH_VERIFICATION_RETRIES {
+        if verify_proof_onchain(aligned_verification_data, chain.clone(), rpc_url)
+            .await
+            .is_ok_and(|r| r)
+        {
+            return Ok(());
+        }
+
+        debug!("Proof not verified yet. Waiting 10 seconds before checking again...");
```

Comment:
```suggestion
        debug!("Proof not verified yet. Waiting {} seconds before checking again...", AWAIT_BATCH_VERIFICATION_TIMEOUT);
```

Reply:
Solved in [617be52](https://github.com/yetanotherco/aligned_layer/pull/583/commits/617be520af0af074a57ce3d3089384b97ebb5293)

Reply:
Solved in [19d1cd8](https://github.com/yetanotherco/aligned_layer/pull/583/commits/19d1cd849d08e9a5e045f20c8c1678eb2f6307dd)